lib/autoyast.pm

```diff
@@ -783,6 +785,30 @@ sub expand_agama_profile {
     return $profile_url;
 }
 
+=head2 generate_json_profile
+
+ generate_json_profile();
+
+ Return the URL of generated JSON profile
+
+=cut
+
+sub generate_json_profile {
+    my $profile_name = "generated_profile.json";
+    my $casedir = get_required_var('CASEDIR');
+
+    my @profile_options = map { " --tla-" . (/true|false/ ? "code" : "str") . " $_" }
+      split(' ', trim(get_var('AGAMA_PROFILE_OPTIONS')));
```

Comment:
According to the VR https://openqa.suse.de/tests/16895737#settings, we need provide variables one by one:`AGAMA_PROFILE_OPTIONS	product="SLES" scripts_post="enable_root_login" bootloader=true` 
Is it possible to get them by get_var()? Because most of them are already defined in openQA variables vars.json.

Reply:
this options will be specific to build the profile, they are simple boolean flags to say the parts you want in your composition, there is not get_var involve or the need for it, they don't need to match any other settings, perhaps only the SLES/SLES_SAP, but still need to think about that. This way to build the profile is totally different than those templates in Perl.